### ISSUE TITLE
question about accuracy

### ISSUE THREAD

**Issue by ayqh** [aymericdamien/TensorFlow-Examples#26]
int "recurrent network" the loss was calculated by the function   tf.nn.softmax_cross_entropy_with_logits  ,and the correct_pred is calculated by tf.equal(tf.argmax(pred,1), tf.argmax(y,1)), which take "pred " directly, how dose this make sense? I think it should be softmax(pred) instead of pred, but I code just works fine. This really confuse me. Can somebody explain whats going on ? thanks!

**Comment by aymericdamien** [comment]
It is working fine because it is using argmax of pred. Last layer of RNN example is a 10 dimension vector (for every label). So the predicted class is represented by that layer highest value index (that you get with argmax).  Softmax is just used to squash values between 0 and 1. 
Softmax is used while training because having a normalized probabilities distribution may help for better performances, but when predicting, softmax is actually not needed.

**Comment by christophetrinh** [comment]
I have a question, i use the same structrure of code as yours, but in order to do detection. However, when i try to add the softmaxlayer "return tf.nn.softmax(tf.matmul(layer_3, _weights['out']) + _biases['out'])", i don't really understand why the cost didn't converge.  thanks!

**Comment by aymericdamien** [comment]
It is hard to say without knowing your data, because network structure is also dependent of your data. If you are parsing words (ids), then your need to add an embedding layer.

**Comment by jakob-grabner** [comment]
I also have a question to recurrent network. The RNN function returns only the last model output. `return tf.matmul(outputs[-1], weights['out']) + biases['out']` which means the loss is also only calculated from the last output. One example in the Tensorflow repo ([ptb_word_lm](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/models/rnn/ptb/ptb_word_lm.py)) uses the every output, not only the output of the last step. What is the right approach, or on what does it depend?

**Comment by aymericdamien** [comment]
It is because in seq2seq is an encoding/decoding process whose output is a sequence so calculate loss for every output is important. However in our example, we are simply doing classification over a whole sequence, so only the last output is meaningful (which is the output after all timesteps have been 'processed').

**Comment by christophetrinh** [comment]
I am very confused with the terms batch_size and n_steps. Does the lstm updates his parameters after n_steps ? In the recurrent network, the lstm is feed with a n_steps list of [batch_size,n_input]. So, in the case of the MNIST classification, the cell is feed with 128 samples of 28pixels every step in range(n_steps) ??
